Add parsing routines for single OPTIMADE JSON

ml-evs opened the pull request:
- Adds parsing and display support for single OPTIMADE JSON structures, including an alpha-quartz (`mp-7000`) sample.
- Reuses shared matrix utilities for coordinate conversion instead of manual calculations.
- Adds tests for OPTIMADE parsing and 3×3 matrix determinants.
- Partial occupancies and structures with incomplete data are not yet supported.

Addresses #99.

ml-evs committed:
Add parsing routines for single OPTIMADE json

- includes test for mp-7000 alpha quartz

janosh commented:
wonderful! thanks @ml-evs 👍  i'll take a look now.

> Had some minor pre-commit issues, ended up restricting the linting to just my changes but still keep seeing:

maybe a `deno` version issue? i'm currently pinned to 2.3.7 due to https://github.com/pngwn/MDsveX/issues/743 (and because of `deno fmt` adds spurious white space in 2.4+)

ml-evs committed:
Linting with deno 2.3.7

janosh committed:
address coderabbit review

janosh committed:
rename math.(transpose_matrix->transpose_3x3_matrix)

janosh committed:
add and test math.det_3x3

janosh committed:
fix failing test from structure.sites.length = undefined when parsing new src/site/structures/mp-7000.json

janosh committed:
refactor OPTIMADE JSON parsing and improve tests

- use matrix ops to simplify coordinate transformation
- remove pointless helper function for building lattice matrix
- update tests to verify lattice matrix and coordinate transformations for non-orthogonal lattices
- add checks for expected xyz coordinates

janosh reviewed (approved):
that was fast! thanks @ml-evs, looks great. ready to go from your end?

ml-evs commented:
Go for it! I'll see if I can get around to the other points in that issue
